jaxb-ri/bundles/ri/src/main/resources/bin/schemagen.sh

```diff
@@ -72,37 +72,71 @@ then
     cd $saveddir
 fi
 
-if [ -n "$CLASSPATH" ] ; then
-  LOCALCLASSPATH="$CLASSPATH"
-fi
+#JXC module path
+JAXB_PATH=${JAXB_HOME}/mod/jaxb-api.jar:\
+${JAXB_HOME}/mod/jaxb-jxc.jar:\
+${JAXB_HOME}/mod/jaxb-xjc.jar:\
+${JAXB_HOME}/mod/jaxb-runtime.jar:\
+${JAXB_HOME}/mod/stax-ex.jar:\
+${JAXB_HOME}/mod/istack-commons-runtime.jar:\
+${JAXB_HOME}/mod/istack-commons-tools.jar:\
+${JAXB_HOME}/mod/FastInfoset.jar:\
+${JAXB_HOME}/mod/dtd-parser.jar:\
+${JAXB_HOME}/mod/rngom.jar:\
+${JAXB_HOME}/mod/codemodel.jar:\
+${JAXB_HOME}/mod/xsom.jar:\
+${JAXB_HOME}/mod/txw2.jar:\
+${JAXB_HOME}/mod/javax.activation.jar
+
 
 # add the api jar file
-if [ -z "$LOCALCLASSPATH" ] ; then
-    LOCALCLASSPATH="$JAXB_HOME"/lib/jaxb-api.jar:"$JAXB_HOME"/lib/jaxb-core.jar:"$JAXB_HOME"/lib/jaxb-xjc.jar:"$JAXB_HOME"/lib/jaxb-jxc.jar:"$JAXB_HOME"/lib/jaxb-impl.jar
+if [ -n ${CLASSPATH} ] ; then
+    LOCALPATH=${JAXB_PATH}:"${CLASSPATH}"
 else
-    LOCALCLASSPATH="$JAXB_HOME"/lib/jaxb-api.jar:"$JAXB_HOME"/lib/jaxb-core.jar:"$JAXB_HOME"/lib/jaxb-xjc.jar:"$JAXB_HOME"/lib/jaxb-jxc.jar:"$JAXB_HOME"/lib/jaxb-impl.jar:"$LOCALCLASSPATH"
+    LOCALPATH=${JAXB_PATH}
 fi
 
 
 if [ -n "$JAVA_HOME" ]
 then
     JAVA="$JAVA_HOME"/bin/java
-    LOCALCLASSPATH="$JAVA_HOME"/lib/tools.jar:"$LOCALCLASSPATH"
 else
     JAVA=java
     JAVACMD=`which $JAVA`
     BINDIR=`dirname $JAVACMD`
-    LOCALCLASSPATH="$BINDIR"/../lib/tools.jar:"$LOCALCLASSPATH"
 fi
 [ `expr \`uname\` : 'CYGWIN'` -eq 6 ] &&
 {
-    LOCALCLASSPATH=`cygpath -w -p ${LOCALCLASSPATH}`
+    LOCALPATH=`cygpath -w -p ${LOCALPATH}`
 }
 
 if [ `expr \`uname\` : 'CYGWIN'` -eq 6 ]
 then
     JAXB_HOME="`cygpath -w "$JAXB_HOME"`"
 fi
 
+VERSION_STRING=`${JAVA} -version 2>&1 | head -n 1 | sed -E 's/.+version "(.+)"/\1/'`
```

Comment:
Any special reason why `java -fullversion` is not used?

Reply:
We are interested only in the "major" version part so -version is sufficient, but I can update to fullversion and remove "head -n 1" than from parsing.